TTYC.IdentityServer/ResourceOwnerPasswordValidator.cs

```diff
@@ -0,0 +1,35 @@
+﻿using IdentityServer4.Models;
+using IdentityServer4.Validation;
+using MediatR;
+using TTYC.Application;
+using TTYC.Application.Users.Queries.ValidatePassword;
+
+namespace TTYC.IdentityServer
+{
+	public class ResourceOwnerPasswordValidator : IResourceOwnerPasswordValidator
+	{
+		private readonly ISender mediatr;
+
+		public ResourceOwnerPasswordValidator(ISender mediatr)
+		{
+			this.mediatr = mediatr;
+		}
+
+		public async Task ValidateAsync(ResourceOwnerPasswordValidationContext context)
+		{
+			var query = new ValidatePasswordQuery()
+			{
+				UserName = context.UserName
+			};
+
+			var user = await mediatr.Send(query);
+            var isMatch = PasswordHelper.VerifyHashedPassword(user.Password, context.Password);
+
+            if (isMatch) {
+				context.Result = new GrantValidationResult(user.Id.ToString(), GrantType.ResourceOwnerPassword);
```

Comment:
Let me show why I asked not to use tabs in code
![image](https://user-images.githubusercontent.com/46454404/185995586-653495ee-2be8-4b0f-bb16-ba6db14c0f27.png)

Reply:
I've changed this for those commit, don't know why it isn't work